README.md

```diff
@@ -300,53 +343,85 @@
   end)
   ```
 
-* <a name="space-before-anonymous-fun-arrow"></a>
-  Always use a space before `->` in 0-arity anonymous functions.
-  <sup>[[link](#space-before-anonymous-fun-arrow)]</sup>
+* <a name="fun-parens"></a>
+  Always use parentheses around arguments to definitions (such as `def`, `defp`, `defmacro`, `defmacrop`, `defdelegate`). Don't omit them even when a function has no arguments.
+  <sup>[[link](#fun-parens)]</sup>
 
   ```elixir
   # Bad
-  Task.async(fn->
-    ExUnit.Diff.script(left, right)
-  end)
+  def main arg1, arg2 do
+    # ...
+  end
+
+  defmacro env do
+    # ...
+  end
 
   # Good
-  Task.async(fn ->
-    ExUnit.Diff.script(left, right)
-  end)
+  def main(arg1, arg2) do
+    # ...
+  end
+
+  defmacro env() do
+    # ...
+  end
   ```
 
-* <a name="pipeline-operator"></a>
-  Favor the pipeline operator `|>` to chain function calls together.
-  <sup>[[link](#pipeline-operator)]</sup>
+* <a name="parens-in-zero-arity-types"></a>
+  Never use parens on zero-arity types.
+  <sup>[[link](#parens-in-zero-arity-types)]</sup>
 
   ```elixir
   # Bad
-  String.downcase(String.strip(input))
+  @spec start_link(module(), term(), Keyword.t()) :: on_start()
 
   # Good
-  input |> String.strip() |> String.downcase()
+  @spec start_link(module, term, Keyword.t) :: on_start
   ```
 
-  For a multi-line pipeline, place each function call on a new line, and retain the level of indentation.
+### Layout
+
+* <a name="no-semicolon"></a>
+  Use one expression per line. Don't use semicolons (`;`) to separate statements and expressions.
+  <sup>[[link](#no-semicolon)]</sup>
 
   ```elixir
-  input
-  |> String.strip()
-  |> String.downcase()
-  |> String.slice(1, 3)
+  # Bad
+  stacktrace = System.stacktrace(); fun.(stacktrace)
+
+  # Good
+  stacktrace = System.stacktrace()
+  fun.(stacktrace)
   ```
 
-* <a name="needless-pipeline"></a>
-  Avoid needless pipelines like the plague.
-  <sup>[[link](#needless-pipeline)]</sup>
+* <a name="multi-line-expr-assignment"></a>
+  When assigning the result of a multi-line expression, begin the expression on a new line.
+  <sup>[[link](#multi-line-expr-assignment)]</sup>
 
   ```elixir
   # Bad
-  result = input |> String.strip()
+  {found, not_found} = files
+                       |> Enum.map(&Path.expand(&1, path))
+                       |> Enum.partition(&File.exists?/1)
+
+  prefix = case base do
+             :binary -> "0b"
+             :octal -> "0o"
+             :hex -> "0x"
+           end
 
   # Good
-  result = String.strip(input)
+  {found, not_found} =
+    files
+    |> Enum.map(&Path.expand(&1, path))
+    |> Enum.partition(&File.exists?/1)
+
+  prefix =
+    case base do
+      :binary -> "0b"
+      :octal -> "0o"
+      :hex -> "0x"
+    end
   ```
 
 * <a name="binary-operators-at-eols"></a>
```

Comment:
When we have a binding or a continuation of the argument construction we don't indent next lines to one more level.